### PR TITLE
UI: Add more auth test coverage

### PR DESCRIPTION
**Pull request by hashishaw**
### Description
This PR adds additional coverage for the auth form, checking that the payload and URL are as expected when all the inputs are filled in.

### CODE REVIEW

File: ui/tests/acceptance/auth-test.js [hunk 1]
```diff
@@ -123,6 +125,138 @@ module('Acceptance | auth', function (hooks) {
     }
   });
 
+  module('it sends the right payload when authenticating', function (hooks) {
```
Comment (hashishaw) [inline ui/tests/acceptance/auth-test.js:128]: This module is basically a more robust version of `it sends the right attributes when authenticating` -- thoughts about replacing that module with this one?

File: ui/tests/acceptance/auth-test.js [hunk 2]
```diff
@@ -123,6 +125,138 @@ module('Acceptance | auth', function (hooks) {
...
+      }`, async function (assert) {
+        const { type } = backend;
+        const expected = this.expected[type];
+        const isOidc = ['oidc', 'jwt'].includes(type);
+        assert.expect(isOidc ? 6 : 2);
+
+        this.assertReq = (req) => {
+          const body = type === 'token' ? req.requestHeaders : JSON.parse(req.requestBody);
+          if (isOidc && !body.role) {
+            // OIDC and JWT auth form calls the endpoint every time the role or mount is updated.
```
Comment (hellobontempo) [inline ui/tests/acceptance/auth-test.js:226]: thank you for adding this more explicit comment!

File: ui/tests/acceptance/auth-test.js [hunk 3]
```diff
@@ -62,62 +62,112 @@ module('Acceptance | auth', function (hooks) {
...
+          Object.keys(expected.payload).forEach((expKey) => {
+            assert.strictEqual(
+              body[expKey],
+              expected.payload[expKey],
+              `${type} payload includes ${expKey} with expected value`
+            );
+          });
         };
         await visit('/vault/auth');
         await component.selectMethod(type);
```
Comment (hellobontempo) [inline ui/tests/acceptance/auth-test.js:158]: should we update these selectors as we can so we can slowly move away from `ember-cli-page-object`?

**Review by hellobontempo — APPROVED** [review]
Thank you for tackling this! Just a question about updating some uses of ember-cli-page-object, non-blocking though

File: ui/tests/acceptance/auth-test.js (hunk 3 above)
Comment (hashishaw) [reply ui/tests/acceptance/auth-test.js:158]: Since we're not backporting this, I think that's a good idea!

**Pull request merged**